programs/cipher/pbcrypt.c

```diff
@@ -0,0 +1,720 @@
+    do {                                                     \
+        if( ! ( expr ) ) {                                   \
+            mbedtls_fprintf( stderr, "Internal error: %s\n", \
+                             #expr );                        \
+            ret = PBCRYPT_ERR;                               \
+            goto exit;                                       \
+        }                                                    \
+    }                                                        \
+    while( 0 )
+
+#define CHECK_FWRITE( data, size, file )                                \
+        do                                                              \
+        {                                                               \
+            size_t CHECK_WRITE__size = fwrite( data, 1, size, file );   \
+            if( CHECK_WRITE__size != size )                             \
+            {                                                           \
+                perror( "fwrite" );                                     \
+                ret = -1;                                               \
+                goto exit;                                              \
+            }                                                           \
+        }                                                               \
+        while ( 0 )
+#define CHECK_FREAD( data, size, file )                                 \
+        do                                                              \
+        {                                                               \
+            size_t CHECK_READ__size = fread( data, 1, size, file );     \
+            if( CHECK_READ__size == (size_t)( -1 ) )                    \
+            {                                                           \
+                perror( "fread" );                                      \
+                ret = -1;                                               \
+                goto exit;                                              \
+            }                                                           \
+            else if( CHECK_READ__size != size )                         \
+            {                                                           \
+                mbedtls_fprintf( stderr,                                \
+                                 "fread: short read at line %d\n",      \
+                                 __LINE__ );                            \
+                ret = PBCRYPT_ERR;                                      \
+                goto exit;                                              \
+            }                                                           \
+        }                                                               \
+        while ( 0 )
+
+/* Implementation that should never be optimized out by the compiler */
+static void mbedtls_zeroize( void *v, size_t n ) {
```

Comment:
Note: in the meantime we introduced `mbedtls_platform_zeroize()`.

Reply:
Not in 2.7, and already noted for the other branches (see PR description).

Reply:
Oh, right. Sorry for the noise then.